akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala

```diff
@@ -89,187 +76,163 @@ private object PoolSlot {
    * Completion and errors from the connection are not surfaced to the outside (unless we are
    * shutting down completely).
    */
-  private class SlotProcessor(slotIx: Int, connectionFlow: Flow[HttpRequest, HttpResponse, Any])(implicit fm: Materializer)
-    extends ActorSubscriber with ActorPublisher[List[ProcessorOut]] with ActorLogging {
-    var exposedPublisher: akka.stream.impl.ActorPublisher[Any] = _
-    var inflightRequests = immutable.Queue.empty[RequestContext]
+  private class SlotProcessor(slotIx: Int, connectionFlow: Flow[HttpRequest, HttpResponse, Any], log: LoggingAdapter)(implicit fm: Materializer)
+    extends GraphStage[FlowShape[SlotCommand, List[ProcessorOut]]] {
 
-    val runnableGraph = Source.actorPublisher[HttpRequest](flowInportProps(self))
-      .via(connectionFlow)
-      .toMat(Sink.actorSubscriber[HttpResponse](flowOutportProps(self)))(Keep.both)
-      .named("SlotProcessorInternalConnectionFlow")
+    val in: Inlet[SlotCommand] = Inlet("SlotProcessor.in")
+    val out: Outlet[List[ProcessorOut]] = Outlet("SlotProcessor.out")
 
-    override def requestStrategy = ZeroRequestStrategy
+    override def shape: FlowShape[SlotCommand, List[ProcessorOut]] = new FlowShape(in, out)
 
-    /**
-     * How PoolProcessor changes its `receive`:
-     * waitingExposedPublisher -> waitingForSubscribePending -> unconnected ->
-     * waitingForDemandFromConnection OR waitingEagerlyConnected -> running
-     * Given slot can become get to 'running' state via 'waitingForDemandFromConnection' or 'waitingEagerlyConnected'.
-     * The difference between those two paths is that the first one is lazy - reacts to DispatchCommand and then uses
-     * inport and outport actors to obtain more items.
-     * Where the second one is eager - reacts to SlotShouldConnectCommand from PoolConductor, sends SlotEvent.ConnectedEagerly
-     * back to conductor and then waits for the first DispatchCommand
-     */
-    override def receive = waitingExposedPublisher
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler { self ⇒
+      private var inflightRequests = immutable.Queue.empty[RequestContext]
```

Comment:
I did benchmarking elsewhere and it does make quite a bit of difference, especially allocations. Also, this will be now more fused than before, so maybe we can have more speed.